src/app/Card.tsx

```diff
@@ -41,27 +42,28 @@ export const Card = (
 
   useEffect(resetActiveHistoryIndex, [history.length]);
 
-  const handleSubmit = async (out: string) => {
-    onSubmit(
-      {
-        input,
-        workingDir: await getWorkingDirectory(),
-        output: JSON.parse(out),
-      },
-      false
-    );
-    resetActiveHistoryIndex();
-  };
+  const handleSubmit = async () => {
+    try {
+      onSubmit(
+        {
+          input,
+          workingDir: await getWorkingDirectory(),
+          output: JSON.parse(await simpleCommandWithResult(id, input)),
+        },
+        false
+      );
+    } catch (error) {
+      onSubmit(
+        {
+          input,
+          workingDir: await getWorkingDirectory(),
+          output: ansiFormat(error as string),
+        },
+        true
+      );
+    }
```

Comment:
How is this? So its clear what can generate an error?

```tsx
const workingDir = await getWorkingDIrectory()
let isError = false
let output: string
try {
    output = JSON.parse(await simpleCommandWithResult(id, input))
     } catch (error) {
    output = ansiFormat(error as string)
    isError = true
}
onSubmit({input, workingDir, output}, isError)
```

That being said, I think `simpleCommandWithResult` should probably just return an enum `CardOutput | CardOutputError` instead of throwing an error.

Reply:
That is nicer, thanks. Done.